### PR TITLE
replaced script tags with template tag and updated related javascript …

### LINKED ISSUE #10033: Replace usage of `text/template` script with `template` element
### Is your proposal related to a problem?

As part of the [Stimulus  🎛️ RFC 78](https://github.com/wagtail/rfcs/pull/78) adoption, it would be good to ensure that all usage of `script` tags that are simply present to represent HTML snippets (templates) be replaced with the `template` element.

This is so that we can avoid using inline scripts and that future Stimulus adoption is easier to migrate towards. 

### Describe the solution you'd like

* Replace the two usages of `<script ... type="text/template"` with simply `<template ...`
* Update the JS behaviour accordingly (see important note in context below)
* Update any Python or JS unit tests that will be impacted by this chance to ensure they pass
* Resolve any edge cases that we encounter with changing from `script` to `template`
* Do not worry about refactoring things in the duplicated JS, in time this will move to Stimulus, for now this is about removing any script usage that should be template usage

#### Current Implementations

* https://github.com/wagtail/wagtail/blob/main/wagtail/documents/static_src/wagtaildocs/js/add-multiple.js#L14
* https://github.com/wagtail/wagtail/blob/main/wagtail/images/static_src/wagtailimages/js/add-multiple.js#L25
* https://github.com/wagtail/wagtail/blob/main/wagtail/images/templates/wagtailimages/multiple/add.html#L49-L74
* https://github.com/wagtail/wagtail/blob/main/wagtail/documents/templates/wagtaildocs/multiple/add.html#L49-L65

#### Important note about using `template`

* Read this before starting https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template
* Most important https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template#avoiding_documentfragment_pitfall
* Correct implementation: `const myClone = template.content.firstElementChild.cloneNode(true);`
* Note: jQuery `.html()` may be enough but do check that event listeners correctly get added

#### Related items

* https://github.com/wagtail/wagtail/issues/10034 (should be done **after** this issue)

### PR DESCRIPTION
**Pull request by rishabhDev2700**
…code

Fixes #10033 

_Please check the following:_

-   [x] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
    -   [x] Run `make lint` from the Wagtail root.
-   [ ] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [x] **Please list the exact browser and operating system versions you tested**:
    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [ ] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

**Comment by rishabhDev2700** [comment]
Development environment: Ubuntu 22.04 64-bit, python 3.10.6

File: wagtail/documents/static_src/wagtaildocs/js/add-multiple.js [hunk 1]
```diff
@@ -11,7 +11,9 @@ $(function () {
     add: function (e, data) {
       var $this = $(this);
       var that = $this.data('blueimp-fileupload') || $this.data('fileupload');
-      var li = $($('#upload-list-item').html()).addClass('upload-uploading');
+      var li = $($('#upload-list-item').content.html()).addClass(
```
Comment (lb-) [inline wagtail/documents/static_src/wagtaildocs/js/add-multiple.js:14]:
```suggestion
      var li = $($('#upload-list-item').html()).addClass(
```

Maybe this can just be `html` without the `.content` - please check how jQuery handles contents within template tags.

**Review by lb- — CHANGES_REQUESTED** [review]
@rishabhDev2700  What browser did you test this in? It is not working for me in Firefox 109.

Screenshot 2023-02-08 at 7 38 51 am

Screenshot 2023-02-08 at 7 38 38 am

**Comment by lb-** [comment]
@rishabhDev2700  I recommend you add screenshots of the behaviour working when this is ready for a review again.

Thank you

**Comment by rishabhDev2700** [comment]
Thanks for your time I will look into this issue in-depth

**Comment by rishabhDev2700** [comment]
Screenshot from 2023-02-08 12-12-12
Screenshot from 2023-02-08 12-12-25
doc1
doc

I am using Firefox 109.0.1 on Ubuntu 22.04.1 LTS. I am not getting any warning or error. The Upload button is working properly with single as well as multiple image/document uploads. On your System were you able to upload files?

**Comment by rishabhDev2700** [comment]
Can there be issue related to wagtail version?
I tried reinstalling the local copy of wagtail
Screenshot from 2023-02-08 12-46-04

**Comment by lb-** [comment]
@rishabhDev2700 it looks like your Dev environment may not be set up correctly. 

If you add a console log statement or maybe just clear some styles and save. Does the front-end code rebuild and then you can see those changes in your local Wagtail setup? (Logging in console, styles changed).

**Comment by rishabhDev2700** [comment]
I didn't rebuild the wagtail after editing. Now I am making changes

**Comment by rishabhDev2700** [comment]
@lb-  One check has failed for test-postgres regarding testing user logout, It was working properly with changes suggested by you. what should I do now?

**Comment by lb-** [comment]
@rishabhDev2700 you can ignore that test. Thanks for checking though. 

It's a test against the future release of Django, just to preempt any problems.

**Comment by rishabhDev2700** [comment]
Are these changes

> @rishabhDev2700 you can ignore that test. Thanks for checking though.
> 
> It's a test against the future release of Django, just to preempt any problems.

Is there anything else that needs to be done regarding the issue I was working on?

**Comment by lb-** [comment]
@rishabhDev2700 thanks. Myself or someone on the core team will review soon. 

Feel free to read up on https://github.com/wagtail/wagtail/issues/10034

Give it a go but please ensure you understand the usage and the code a bit before claiming it. Also, it's ok to ask lots of questions on the issue if it's not clear enough. 

Thanks.

**Review by lb- — APPROVED** [review]
@rishabhDev2700 thanks for giving this another go and working through the dev environment issues.

I have validated this and it's good to go. Will merge in.

I will add you to the contributors as `Rishabh Kumar Bahukhandi` but let me know if you want a different name shown there.
